packages/react/src/time-input/docs/overview.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: Time input
+description: The Time input component allows users to enter a time free-form, and in various formats.
+group: Forms
+storybookPath: /story/Forms-TimeInput--basic
+unreleased: true
+figmaGalleryNodeId: REPLACE_ME
+relatedComponents: ['text-input']
+---
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState({ raw: '' });
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = Boolean(value.raw) && !isValidTime(value.raw);
+
+	return (
+		<TimeInput
+			invalid={invalid}
+			label="Time"
+			message="Enter a valid time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+Time input is superset of [Text input](/components/text-input), with additional functionality specific to entering times. It accepts all of Text input’s props, with changes to `onChange` and `value`.
+
+Time input is a controlled component which means consumers of this component need to manage the state of this component by using the `value` and `onChange` props.
+
+The `onChange` handler provides an object with both a `raw` and `formatted` version of the time. `raw` is always in `'HH:mm'` and `formatted` is provided in the `timeFormat` used. You can use this to display the time elsewhere in the UI, where necessary.
+
+<DoHeading />
+
+- use when users are specifically required to enter a specific time.
+
+<DontHeading />
+
+- use when users are required to enter free-form text that isn’t a time.
+
+## Invalid
+
+Use the `invalid` prop to indicate if the user input is invalid. The `isValidTime()` function can be used to help identify whether the input is valid.
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState({ raw: '9:66' });
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = Boolean(value.raw) && !isValidTime(value.raw);
+
+	return (
+		<TimeInput
+			invalid={invalid}
+			label="Time"
+			message="Enter a valid time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+## Changing the time format
+
+The Time input component allows users to enter dates in various formats. To input 9:00 pm, a user could input '9pm', '9:00pm', '2100', '21:00', or other minor variations on these formats.
```

Comment:
> a user could input '9pm', '9:00pm', '2100', '21:00', or other minor variations on these formats

We should have a unit tests testing for as many variations of times as we can think of.

ISO format is a type of time that someone would likely want to paste into the time input.

Reply:
I believe I do?

Reply:
@Dan503 do we have an example of where someone would get an ISO date string from? I can't think of any off the top of my head 🤷

Reply:
@ChrisLaneAU this is a pretty easy way to generate an iso string
```js
new Date().toISOString() // => '2024-04-04T03:32:49.648Z'
```

Reply:
Yes, you're spot on. My question is more about where someone would copy/paste this from?

Reply:
Maybe someone has a spreadsheet full of values that was spat out from a database and they need to copy paste the database values from the spreadsheet to an assessment.

Reply:
This could happen, it's true. If it does, these folks could transform the date in their spreadsheet. I think supporting ISO strings would add complexity that there is no definitive need for. At least not yet. I'd be comfortable keeping this edge-case aside and returning to it later, if needed.

It's always a hard balance of how to serve our users best with our limited time. For me, I see a lot of these edge-cases taking time away from other, user-requested tasks and I tend to draw a line before we get too deep into diminishing returns. Having said this, I'm happy to be overruled, @stowball, if you see value, particularly as you are the PR owner 🙂

Reply:
I won't add support for date strings (it's a time input) but I have added support for more formats [`9f04139` (#1582)](https://github.com/agriculturegovau/agds-next/pull/1582/commits/9f041393d7d5431ec7b0b054d4fa17df8a46bef5)